contracts/reserve/ExchangeHelper.sol

```diff
@@ -317,4 +317,51 @@ contract ExchangeHelper is DAOUpgradeableContract {
 		cDai.transfer(address(reserve), cDaiInput);
 		return reserve.buy(cDaiInput, _minReturn, _targetAddress);
 	}
+
+	/**
+	@dev Helper to swap tokens in the Uniswap
+	*@param _uniswapContract Uniswap Router Contract
+	*@param _token token to buy or sell for
+	*@param _tokenAmount token amount to sell or buy
+	*@param _isBuy if swap transaction to buy DAI or sell DAI
+	*@param _minDAIAmount minimum DAI amount to get in swap transaction if transaction is buy
+	*@param _minTokenReturn minimum token amount to get in swap transaction if transaction is sell
+	*@param _receiver receiver of tokens after swap transaction
+	 */
+	function _uniswapSwap(
+		Uniswap _uniswapContract,
```

Comment:
this shouldnt be a parameter, in the helper fetch it from nameservice

Reply:
I needed it in the buy function itself so did not want to call nameservice twice in order to gas save but if you think it should also get from nameService I can change

Reply:
Moved to helper